src/main/java/com/guidetrack/mentorship_tracker/config/ApplicationConfig.java

```diff
@@ -0,0 +1,40 @@
+package com.guidetrack.mentorship_tracker.config;
+
+import com.guidetrack.mentorship_tracker.repositories.BaseUserRepository;
+import com.guidetrack.mentorship_tracker.services.impl.UserDetailsServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfig {
+    private final BaseUserRepository baseUserRepository;
+     @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    public UserDetailsServiceImpl getUserService() {
```

Comment:
Firstly you are using a concrete class.
Secondly, there is no need to create a bean of this,
Simply use it through dependency injection in the authentication provider method